kedro-telemetry/tests/test_masking.py

```diff
@@ -64,14 +64,14 @@ def fake_metadata(fake_root_dir):
 class TestCLIMasking:
     def test_get_cli_structure_raw(self, mocker, fake_metadata):
         Module = namedtuple("Module", ["cli"])
-        mocker.patch(
-            "kedro.framework.cli.cli.importlib.import_module",
-            return_value=Module(cli=cli),
-        )
         mocker.patch("kedro.framework.cli.cli._is_project", return_value=True)
         mocker.patch(
             "kedro.framework.cli.cli.bootstrap_project", return_value=fake_metadata
         )
+        mocker.patch(
+            "kedro.framework.cli.cli.importlib.import_module",
+            return_value=Module(cli=cli),
+        )
```

Comment:
What is different in python 3.11 that required moving this down?

Reply:
This fix from the 3.11 PR on kedro seemed to fix the failing tests. @SajidAlamQB do you know more about this?

Reply:
I believe it was because it would makes all subsequent mocker.patch fail this could be because of a change in how `unittest.mock` or `importlib` work python 3.11. @noklam do you have any other thoughts on it?